staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -432,11 +432,20 @@ func (r *Request) Body(obj interface{}) *Request {
 		}
 		glogBody("Request Body", data)
 		r.body = bytes.NewReader(data)
+		if len(r.headers.Get("Content-Type")) == 0 && len(r.c.content.ContentType) > 0{
+			r.SetHeader("Content-Type", r.c.content.ContentType)
+		}
 	case []byte:
 		glogBody("Request Body", t)
 		r.body = bytes.NewReader(t)
+		if len(r.headers.Get("Content-Type")) == 0 && len(r.c.content.ContentType) > 0{
+			r.SetHeader("Content-Type", r.c.content.ContentType)
+		}
 	case io.Reader:
 		r.body = t
+		if len(r.headers.Get("Content-Type")) == 0 && len(r.c.content.ContentType) > 0{
+			r.SetHeader("Content-Type", r.c.content.ContentType)
+		}
```

Comment:
@smarterclayton I think this is a reasonable default behavior, and clients that want different behavior can explicitly call `r.SetHeader("Content-Type", ...)` to get the behavior they want (before or after calling `Body`). What do you think?